### PR TITLE
feat(vehicle): LSC (Los Santos Customs) tab 

### LINKED ISSUE #173: Mobile LS Customs (Vehicle Editor while not at LS Customs)
**Is your feature request related to a problem? Please describe.**
No, there is no problem. Just would love to edit the vehicle I am driving on-the-go.

**Describe the solution you'd like**
Essentially just asking for a submenu that will just allow for editing the vehicle.

**Provide Reasoning**
You guys are experienced developers and I am sure I am not the only one that would love this.

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by mentolixite**
Closes #173.

### CODE REVIEW

File: BigBaseV2/src/views/vehicle/view_lsc.cpp [hunk 1]
```diff
@@ -0,0 +1,443 @@
...
+	{
+		static Vehicle player_vehicle{};
+		static std::vector<std::string> slot_display_names{};
+		static std::map<int, int> owned_mods{};
+		static std::map<int, std::vector<std::string>> mod_display_names{};
+		static int selected_slot = -1;
+		static bool can_tires_burst{}, tiresmoke{}, turbo{}, xenon{};
+		static int primary_color{}, secondary_color{}, pearlescent{}, wheel_color{}, interior_color{}, dashboard_color{};
+
+		if (!PED::IS_PED_IN_ANY_VEHICLE(PLAYER::PLAYER_PED_ID(), FALSE))
```
Comment (maybegreat48) [inline BigBaseV2/src/views/vehicle/view_lsc.cpp]: Don't call natives in gui thread (try replacing it with `self::veh == 0`)

Comment (mentolixite) [reply BigBaseV2/src/views/vehicle/view_lsc.cpp]: Thanks!

**Comment by Yimura** [comment]
You're using a lot of natives in the UI thread, @maybegreat48 mentioned this is detected somewhere, as you're not running them in a game script. You'll have to make some kind of service that runs next to your UI to cache vehicle information and then render it.

**Comment by Yimura** [comment]
You're using a lot of natives in the UI thread, @maybegreat48 mentioned this is detected somewhere as you're not running them in a game script. You'll have to make some kind of service that runs next to your UI to cache vehicle information and then render it.

**Comment by mentolixite** [comment]
Calling natives at all or a lot of them subsequently?

https://github.com/Yimura/YimMenu/blob/8f545c48565105ab3eb2156a2bfbbe96562b3aba/BigBaseV2/src/views/players/view_player.cpp#L105

**Comment by Yimura** [comment]
Calling natives at all, also button components are wrapped in a game script callback so it's fine to use natives there.

**Comment by mentolixite** [comment]
As far as I can see all natives calls are in the fiber queue, did I miss any?

**Comment by Yimura** [comment]
Yeah everything appears to be fine with your code, will have to look at it locally though, it's a lot of code so I can't just approve it like this.

**Comment by mentolixite** [comment]
No problem.

Please note that:

1. @GibsonHF is the author, I forgot to set him as author when I force-pushed.
2. #253 also affects this.

**Review by Yimura — APPROVED** [review]

**Comment by Yimura** [comment]
I didn't test this thoroughly but the code is nicely split up and if any bugs come from adding this feature people can just report them by creating an issue.

Good work :+1:

**Pull request merged**